src/nbl/asset/utils/CGeometryCreator.cpp

```diff
@@ -398,258 +399,615 @@ core::smart_refctd_ptr<ICPUPolygonGeometry> CGeometryCreator::createSphere(float
 		// we don't start at 0.
 
 		double ay = 0;//AngleY / 2;
-
-		using quant_normal_t = CQuantNormalCache::value_type_t<EF_A2B10G10R10_SNORM_PACK32>;
-		uint8_t* tmpMemPtr = tmpMem;
-		for (uint32_t y = 0; y < polyCountY; ++y)
-		{
-			ay += AngleY;
-			const double sinay = sin(ay);
-			axz = 0;
-
-			// calculate the necessary vertices without the doubled one
-			uint8_t* oldTmpMemPtr = tmpMemPtr;
-			for (uint32_t xz = 0; xz < polyCountX; ++xz)
-			{
-				// calculate points position
-
-				float32_t3 pos(static_cast<float>(cos(axz) * sinay),
-					static_cast<float>(cos(ay)),
-					static_cast<float>(sin(axz) * sinay));
-				// for spheres the normal is the position
-				core::vectorSIMDf normal(&pos.X);
-				normal.makeSafe3D();
-				quant_normal_t quantizedNormal = quantNormalCache->quantize<EF_A2B10G10R10_SNORM_PACK32>(normal);
-				pos *= radius;
-
-				// calculate texture coordinates via sphere mapping
-				// tu is the same on each level, so only calculate once
-				float tu = 0.5f;
-				//if (y==0)
-				//{
-				if (normal.Y != -1.0f && normal.Y != 1.0f)
-					tu = static_cast<float>(acos(core::clamp(normal.X / sinay, -1.0, 1.0)) * 0.5 * core::RECIPROCAL_PI<double>());
-				if (normal.Z < 0.0f)
-					tu = 1 - tu;
-				//}
-				//else
-					//tu = ((float*)(tmpMem+(i-polyCountXPitch)*vertexSize))[4];
-
-				((float*)tmpMemPtr)[0] = pos.X;
-				((float*)tmpMemPtr)[1] = pos.Y;
-				((float*)tmpMemPtr)[2] = pos.Z;
-				((float*)tmpMemPtr)[4] = tu;
-				((float*)tmpMemPtr)[5] = static_cast<float>(ay * core::RECIPROCAL_PI<double>());
-				((quant_normal_t*)tmpMemPtr)[6] = quantizedNormal;
-				static_assert(sizeof(quant_normal_t)==4u);
-
-				tmpMemPtr += vertexSize;
-				axz += AngleX;
-			}
-			// This is the doubled vertex on the initial position
-
-			((float*)tmpMemPtr)[0] = ((float*)oldTmpMemPtr)[0];
-			((float*)tmpMemPtr)[1] = ((float*)oldTmpMemPtr)[1];
-			((float*)tmpMemPtr)[2] = ((float*)oldTmpMemPtr)[2];
-			((float*)tmpMemPtr)[4] = 1.f;
-			((float*)tmpMemPtr)[5] = ((float*)oldTmpMemPtr)[5];
-			((uint32_t*)tmpMemPtr)[6] = ((uint32_t*)oldTmpMemPtr)[6];
-			tmpMemPtr += vertexSize;
-		}
+		auto vertex_i = 0;
+    for (uint32_t y = 0; y < polyCountY; ++y)
+    {
+      ay += AngleY;
+      const double sinay = sin(ay);
+      axz = 0;
+
+      // calculate the necessary vertices without the doubled one
+			const auto old_vertex_i = vertex_i;
+      for (uint32_t xz = 0; xz < polyCountX; ++xz)
+      {
+        // calculate points position
+
+        float32_t3 pos(static_cast<float>(cos(axz) * sinay),
+          static_cast<float>(cos(ay)),
+          static_cast<float>(sin(axz) * sinay));
+        // for spheres the normal is the position
+        core::vectorSIMDf normal(&pos.x);
+        normal.makeSafe3D();
+        const auto quantizedNormal = quantNormalCache->quantize<NormalCacheFormat>(normal);
```

Comment:
can you make the quantization cache work with `hlsl::` vector types instead of `core::vectorSIMD` (I'm trying to deprecate them)